src/commands/Minion/blastfurnace.ts

```diff
@@ -0,0 +1,181 @@
+			);
+		}
+
+		if (!bar) {
+			return msg.send(
+				`Thats not a valid bar to smelt. Valid bars are ${Smithing.BlastableBars.map(
+					bar => bar.name
+				).join(', ')}.`
+			);
+		}
+
+		if (msg.author.skillLevel(SkillsEnum.Smithing) < bar.level) {
+			return msg.send(
+				`${msg.author.minionName} needs ${bar.level} Smithing to smelt ${bar.name}s.`
+			);
+		}
+
+		let timeToSmithSingleBar = bar.timeToUse + Time.Second / 10;
+
+		// check if they have a coal bag
+		let coalbag = '';
+		if (
+			msg.author.hasItemEquippedOrInBank(itemID('Coal bag')) &&
+			(bar.id === itemID('Steel Bar') ||
+				bar.id === itemID('Mithril Bar') ||
+				bar.id === itemID('Adamantite Bar') ||
+				bar.id === itemID('Runite Bar'))
+		) {
+			coalbag = `\n\n**Boosts:** 60% speed boost for coal bag.`;
+			timeToSmithSingleBar *= 0.625;
+		}
+		let graceful = '';
+		if (!hasGracefulEquipped(msg.author.getGear('skilling'))) {
+			timeToSmithSingleBar *= 1.075;
+			graceful = `\n-7.5% time penalty for not having graceful equipped.`;
+		}
+
+		const maxTripLength = msg.author.maxTripLength(Activity.Smithing);
+
+		// If no quantity provided, set it to the max.
+		if (quantity === null) {
+			quantity = Math.floor(maxTripLength / timeToSmithSingleBar);
+		}
+
+		await msg.author.settings.sync(true);
+		const userBank = msg.author.settings.get(UserSettings.Bank);
+
+		// Check the user has the required ores to smith these bars.
+		// Multiplying the ore required by the quantity of bars.
+		const requiredOres: [string, number][] = Object.entries(bar.inputOres);
+		for (const [oreID, qty] of requiredOres) {
+			if (!bankHasItem(userBank, parseInt(oreID), qty * quantity)) {
+				return msg.send(`You don't have enough ${itemNameFromID(parseInt(oreID))}.`);
+			}
+		}
+
+		const duration = quantity * timeToSmithSingleBar;
+		if (duration > maxTripLength) {
+			return msg.send(
+				`${msg.author.minionName} can't go on trips longer than ${formatDuration(
+					maxTripLength
+				)}, try a lower quantity. The highest amount of ${
+					bar.name
+				}s you can smelt is ${Math.floor(maxTripLength / timeToSmithSingleBar)}.`
+			);
+		}
+
+		// cost to pay the foreman to use blast furance
+		const itemsToRemove = new Bank();
+		const coinstoremove = Math.floor(72000 * (duration / Time.Hour));
+		const gp = msg.author.settings.get(UserSettings.GP);
+		if (gp < coinstoremove) {
+			return msg.send(`You need atleast ${coinstoremove} GP to work at the Blast Furnace.`);
+		}
+
+		itemsToRemove.add('Coins', coinstoremove);
+		await msg.author.removeItemsFromBank(itemsToRemove.bank);
+
+		// Remove the ores from their bank.
+		let newBank: ItemBank = { ...userBank };
+		for (const [oreID, qty] of requiredOres) {
+			if (newBank[parseInt(oreID)] < qty) {
+				this.client.wtf(
+					new Error(`${msg.author.sanitizedName} had insufficient ores to be removed.`)
+				);
+				return;
+			}
+			newBank = removeItemFromBank(newBank, parseInt(oreID), qty * quantity);
+		}
```

Comment:
This isn't something we do anymore as its the old and bad way of doing it, you need to construct a `Bank` and `user.removeItemsFromBank(Bank)`, rather than copying their bank and ovewriting here as youre doing.